### DISCUSSION TITLE
Help using x-select async-data from api and search

### DISCUSSION THREAD

**Discussion by slossetti** [wireui/wireui#789] [Q&A] [answered]
Hi Guys,

I'm trying to get some data from an api to my x-select using async-data in that way:

` <x-select 
      placeholder="Select name"
      :searchable="true"
      :min-items-for-search="1"
      :async-data="[
          'api' => 'https://jsonplaceholder.typicode.com/users',
          'method' => 'GET',
      ]"
      option-label="username"
      option-value="id"
    />`
    
    the info shows correctly but the search is not working. I don't why, because in others x-select using collection was worked.
    
![image](https://github.com/wireui/wireui/assets/57014758/b1d8319c-5988-4448-ba26-0a2e27247bf3)

Can you help me?

**Comment by ph7jack** [comment, marked as answer]
Hello @slossetti 
As documented [here](https://v1.wireui.dev/docs/select), your endpoint should be able to search.
Make sure your API is able to search and retrieve records. 
The select will never "search" using the async search, your endpoint should search and return the data.

**Reply by slossetti** [reply]
Thanks!!! 

Another way could by store into a variable the api results in my controller and then pass it as collection to the x-select?

**Reply by ph7jack** [reply]
Yes, you can pass a collection/array inside the options attribute. 

`<x-select :options="$options" />`
